Gems/BarrierInput/Code/Source/BarrierInputKeyboard.cpp

```diff
@@ -0,0 +1,258 @@
+            ThreadSafeQueueRawTextEvent(text);
+        }
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    void InputDeviceKeyboardBarrier::ThreadSafeQueueRawKeyEvent(uint32_t scanCode, bool rawKeyState)
+    {
+        // From observation, Barrier scan codes in the:
+        // - Range 0x0-0x7F (0-127) correspond to windows scan codes without the extended bit set
+        // - Range 0x100-0x17F (256-383) correspond to windows scan codes with the extended bit set
+        const InputChannelId* inputChannelId = nullptr;
+        if (scanCode < InputChannelIdByScanCodeTable.size())
+        {
+            inputChannelId = InputChannelIdByScanCodeTable[scanCode];
+        }
+        else if (0 <= (scanCode - 0x100) && scanCode < InputChannelIdByScanCodeWithExtendedPrefixTable.size())
+        {
+            inputChannelId = InputChannelIdByScanCodeWithExtendedPrefixTable[scanCode - 0x100];
+        }
+
+        if (inputChannelId)
+        {
+            AZStd::lock_guard<AZStd::mutex> lock(m_threadAwareRawKeyEventQueuesByIdMutex);
+            m_threadAwareRawKeyEventQueuesById[*inputChannelId].push_back(rawKeyState);
+        }
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    void InputDeviceKeyboardBarrier::ThreadSafeQueueRawTextEvent(const AZStd::string& textUTF8)
+    {
+        AZStd::lock_guard<AZStd::mutex> lock(m_threadAwareRawTextEventQueueMutex);
+        m_threadAwareRawTextEventQueue.push_back(textUTF8);
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    char InputDeviceKeyboardBarrier::TranslateRawKeyEventToASCIIChar(uint32_t scanCode,
+                                                                     ModifierMask activeModifiers)
+    {
+        // Map ASCII character pairs keyed by their keyboard scan code, assuming an ANSI mechanical
+        // keyboard layout with a standard QWERTY key mapping. The first element of the pair is the
+        // character that should be produced if the key is pressed while no shift or caps modifiers
+        // are active, while the second element is the character that should be produced if the key
+        // is pressed while a shift or caps modifier is active. Required because Barrier only sends
+        // raw key events, not translated text input. While we would ideally support the full range
+        // of UTF-8 text input, that is beyond the scope of this debug/development only class. Note
+        // that this function assumes an ANSI mechanical keyboard layout with a standard QWERTY key
+        // mapping, and will not produce correct results if used with other key layouts or mappings.
+        // Using std::unordered_map/pair instead of AZStd::unordered_map to avoid allocator issues.
+        static const std::unordered_map<AZ::u32, std::pair<char, char>> ScanCodeToASCIICharMap =
```

Comment:
There is a `AZStd::fixed_unordered_map` which doesn't allocate any memory

Reply:
Updated, thanks for the suggestion, and for the help adding the new constructor to AZStd::fixed_unordered_map that accepts the initializer list!